drivers/counter/counter_gecko_stimer.c

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright (c) 2021, Sateesh Kotapati
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT silabs_gecko_stimer
+
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <soc.h>
+#include <em_cmu.h>
+
+#include "sl_sleeptimer.h"
+#include "sl_atomic.h"
+
+#include <zephyr/drivers/counter.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(counter_gecko, CONFIG_COUNTER_LOG_LEVEL);
+
+#define STIMER_MAX_VALUE       (_RTCC_CNT_MASK)
+#define STIMER_ALARM_NUM        2
+
+struct counter_gecko_config {
+	struct counter_config_info info;
+	void (*irq_config)(void);
+	uint32_t prescaler;
+};
+
+struct counter_gecko_alarm_data {
+	counter_alarm_callback_t callback;
+	uint8_t chan_id;
+	uint32_t ticks;
+	struct device *dev;
+	void *user_data;
+};
+
+struct counter_gecko_top_data {
+	counter_top_callback_t callback;
+	uint32_t ticks;
+	struct device *dev;
+	void *user_data;
+};
+
+struct counter_gecko_data {
+	struct counter_gecko_alarm_data alarm[STIMER_ALARM_NUM];
+	struct counter_gecko_top_data top_data;
+};
+
+static sl_sleeptimer_timer_handle_t alarm_timer[STIMER_ALARM_NUM];
+static sl_sleeptimer_timer_handle_t top_timer;
+
+#ifdef CONFIG_SOC_GECKO_HAS_ERRATA_RTCC_E201
+#define ERRATA_RTCC_E201_MESSAGE					     \
+	"Errata RTCC_E201: In case RTCC prescaler != 1 the module does not " \
+	"reset the counter value on CCV1 compare."
+#endif
+
+static void alarm_callback(sl_sleeptimer_timer_handle_t *handle, void *data)
+{
+	struct counter_gecko_alarm_data *alarm_data = (struct counter_gecko_alarm_data *)data;
+	uint32_t count = ((sl_sleeptimer_get_tick_count()) %
+		(((struct counter_gecko_data *const)(alarm_data->dev)->data)->top_data.ticks));
+
+	if (alarm_data->callback != NULL) {
+		alarm_data->callback(alarm_data->dev, alarm_data->chan_id, count,
+			((struct counter_alarm_cfg *)(alarm_data->user_data))->user_data);
+	}
+}
+
+static void top_callback(sl_sleeptimer_timer_handle_t *handle, void *data)
+{
+	struct counter_gecko_top_data *top_data = (struct counter_gecko_top_data *)data;
+
+	if (top_data->callback != NULL) {
+		top_data->callback(top_data->dev,
+			((struct counter_top_cfg *)(top_data->user_data))->user_data);
+	}
+}
+
+static int counter_gecko_get_value(const struct device *dev, uint32_t *ticks)
+{
+	struct counter_gecko_data *const dev_data = (struct counter_gecko_data *const)(dev)->data;
+
+	*ticks = ((sl_sleeptimer_get_tick_count()) % (dev_data->top_data.ticks));
+
+	return 0;
+}
+
+static int counter_gecko_start(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return 0;
+}
+
+static int counter_gecko_stop(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return 0;
```

Comment:
`-ENOTSUP`?

Reply:
Implemented proper `counter_stop` and `counter_start` functions.